hardcode-migrations.js

```diff
@@ -0,0 +1,7 @@
+// knex migrations use dynamic requires which break with webpack.
+// This hack makes the migrations directory a constant so that at least we can use webpack contexts for the
```

Comment:
Knex has a few calls to `require` where the argument is variable: https://github.com/tgriesser/knex/blob/0.12.3/src/migrate/index.js#L317 - these don't get along with webpack because webpack doesn't know where to look for the files to load. If the argument to require is prefixed with a specific path, then webpack can take all of the js files inside that folder and build an object that can be indexed when calling require.
I couldn't find a better way to do this really, and I agree it's fragile (but knex is fixed at "~0.12.3" so we'd have to be careful when upgrading it but we don't risk it breaking accidentally). Happy to hear any other ideas you may have.

Reply:
You can use https://webpack.js.org/plugins/context-replacement-plugin/ to restrict the dynamically inferred context, I'm not sure how easy it would be in this case because it seems that the start of the path could be anything, but that would be the more standard way to do it

Reply:
We *are* actually using ContextReplacementPlugin. This hack is the only way I found so far to make it work, as it needs a `resourceRegexp` to match the first part of the required path, and the knex code doesn't fit that structure.

Reply:
I even tried to make Webpack ignore those requires and leave them as they are, but Webpack insisted on replacing them with "missing dependency" error messages.